Consul/Operator.cs

```diff
@@ -100,6 +100,37 @@ public class KeyringResponse
         public int NumNodes { get; set; }
     }
 
+    public class Area
+    {
+        /// <summary>
+        /// PeerDatacenter is the peer Consul datacenter that will make up the
+        /// other side of this network area. Network areas always involve a pair
+        /// of datacenters: the datacenter where the area was created, and the
+        /// peer datacenter. This is required.
+        /// </summary>
+        public string PeerDatacenter { get; set; }
+
+        /// <summary>
+        /// RetryJoin specifies the address of Consul servers to join to, such as
+	    /// an IPs or hostnames with an optional port number. This is optional.
+        /// </summary>
+        public string[] RetryJoin { get; set; }
+
+        /// <summary>
+        /// UseTLS specifies whether gossip over this area should be encrypted with TLS
+        /// if possible.
+        /// </summary>
+        public bool UseTLS { get; set; }
+    }
+
+    public class AreaResponse : Area
```

Comment:
Inheritance is not needed here, please remove it.

Reply:
Thanks, @marcin-krystianc  Initially, I considered using inheritance because, in certain cases like the[ List Specific Network Area ](https://developer.hashicorp.com/consul/api-docs/v1.11.x/operator/area#list-specific-network-area) response, additional properties such as PeerDatacenter and RetryJoin are returned. I thought that inheriting from the Area class would make the response more flexible for these scenarios.

Reply:
> Thanks, @marcin-krystianc Initially, I considered using inheritance because, in certain cases like the[ List Specific Network Area ](https://developer.hashicorp.com/consul/api-docs/v1.11.x/operator/area#list-specific-network-area) response, additional properties such as PeerDatacenter and RetryJoin are returned. I think inheriting from the Area class will make the response more flexible for these scenarios.

Ah, I see. It is tempting to reuse existing structure definitions for new ones to save some code, but in my opinion, it isn't always the right thing to do. Given that apart from `Create Network Area` there are other methods like `List Network Areas`, `Update Network Area`, and others - we need to plan how we are going to define requests and responses for all of them. 
My initial plan was to create a dedicated request and response structure definition for each method. That would give us full control over those structures and would define the interface very clearly from a user perspective (It is more code in the library to write but it is very straightforward).
On the other hand, I looked at the consul implementation of Operator_Area API (https://github.com/hashicorp/consul/blob/main/api/operator_area.go) and maybe it is ok to do it similarly.
It would require fewer structure definitions but I guess the API will be less clear from the user perspective (for example `AreaUpdate` methods take `ID` and `Area` objects, but `Area` has already an `ID` but it is not used).

I think I'll leave it to you to decide. Both approaches have pros and cons

Reply:
Thanks for the insights @marcin-krystianc ! I initially removed the inheritance and made adjustments based on your initial recommendations, but since you're leaving it up to me, I've added it back. Please let me know what you think!

Reply:
The use of inheritance somewhat bothers me, but maybe unnecessarily :-) The only suggestion I have is to change the names: `Area` -> `AreaRequest`  and `AreaResponse` -> `Area`